### PR TITLE
Add `EC2CreateInstanceOperator`, `EC2TerminateInstanceOperator`

### PR DESCRIPTION
**Pull request by syedahsn**
This PR creates two new operators `EC2CreateInstanceOperator` and `EC2TerminateInstanceOperator` which will be used to create and terminate EC2 instances respectively. The system test has been updated to use the new operators.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/amazon/aws/operators/ec2.py
Comment (josh-fell) [inline airflow/providers/amazon/aws/operators/ec2.py]: Technically this could start multiple instances based on the `max_` and `min_count` args. Let's update this description to account for that behavior too.

File: airflow/providers/amazon/aws/operators/ec2.py
Comment (josh-fell) [inline airflow/providers/amazon/aws/operators/ec2.py]: Same, related comment here too. This operator _could_ terminate multiple instances. I think users might like to know that upfront if they read the Python API docs instead of having to read the source code.

File: airflow/providers/amazon/aws/operators/ec2.py
Comment (josh-fell) [inline airflow/providers/amazon/aws/operators/ec2.py:150]: Curious to know what a use case would be to have `wait_for_completion` as a template field? What did you have in mind here?

File: airflow/providers/amazon/aws/operators/ec2.py
Comment (josh-fell) [inline airflow/providers/amazon/aws/operators/ec2.py]: This would unpack a single `instance_id` into each character individually. Probably should check the type of `instance_id` first (i.e. `str` or `list`) and normalize to a list of `instance_id`s.
```suggestion
        self.instance_ids = [instance_id] if isinstance(instance_id, str) else instance_id
```

**Review by josh-fell — CHANGES_REQUESTED** [review]
Pretty solid, just some small comments but one that should be addressed related to unpacking an arg.

File: tests/providers/amazon/aws/operators/test_ec2.py
Comment (josh-fell) [inline tests/providers/amazon/aws/operators/test_ec2.py:46]: Might be wise to add a test for creating multiple instances too. WDYT?

File: tests/providers/amazon/aws/operators/test_ec2.py
Comment (josh-fell) [inline tests/providers/amazon/aws/operators/test_ec2.py]:
```suggestion
    def test_terminate_instance(self):
```

File: tests/providers/amazon/aws/operators/test_ec2.py
Comment (josh-fell) [inline tests/providers/amazon/aws/operators/test_ec2.py:87]: Same here. WDYT about adding a test for termination multiple instances?

File: airflow/providers/amazon/aws/operators/ec2.py
Comment (syedahsn) [reply airflow/providers/amazon/aws/operators/ec2.py]: Good point, I'll update the description. Thanks!

File: airflow/providers/amazon/aws/operators/ec2.py
Comment (syedahsn) [reply airflow/providers/amazon/aws/operators/ec2.py:150]: I was thinking of a situation where someone wrote a task (say as a TaskFlow task) that would determine if they wanted to wait for the instances to be completed before continuing. Not sure if that is a valid case or not. Is there any harm in having extra templated fields?

Comment (josh-fell) [reply airflow/providers/amazon/aws/operators/ec2.py:150]: Just curious. I don't recall typically seeing booleans as template fields.

File: airflow/providers/amazon/aws/operators/ec2.py
Comment (josh-fell) [inline airflow/providers/amazon/aws/operators/ec2.py]:
```suggestion
    :param instance_ids: ID of the instance to be terminated.
```
Just so this parameter is properly displayed in the Python API docs.

File: airflow/providers/amazon/aws/operators/ec2.py
Comment (uranusjr) [reply airflow/providers/amazon/aws/operators/ec2.py:150]: Currently template fields are also used for XComArg, where boolean makes sense. There’re talks to separate those two ideas but for now having the field can be useful.

Comment (eladkal) [reply airflow/providers/amazon/aws/operators/ec2.py:150]: Do we have an issue for this task?

**Review by josh-fell — APPROVED** [review]
Small docstring update, but LGTM!

**Review by eladkal — APPROVED** [review]
LGTM
will merge when CI is green

**Comment by josh-fell** [comment]
Seems like the "Wait for CI images" step failure could be related?
```
ERROR tests/system/providers/amazon/aws/example_ec2.py - botocore.exceptions.NoCredentialsError: Unable to locate credentials
```
```
______ ERROR collecting tests/system/providers/amazon/aws/example_ec2.py _______
tests/system/providers/amazon/aws/example_ec2.py:91: in <module>
    image_id = _get_latest_ami_id()
tests/system/providers/amazon/aws/example_ec2.py:52: in _get_latest_ami_id
    Filters=[{"Name": "description", "Values": [image_prefix]}], Owners=["amazon"]
/usr/local/lib/python3.7/site-packages/botocore/client.py:530: in _api_call
    return self._make_api_call(operation_name, kwargs)
/usr/local/lib/python3.7/site-packages/botocore/client.py:944: in _make_api_call
    operation_model, request_dict, request_context
/usr/local/lib/python3.7/site-packages/botocore/client.py:966: in _make_request
    return self._endpoint.make_request(operation_model, request_dict)
/usr/local/lib/python3.7/site-packages/botocore/endpoint.py:119: in make_request
    return self._send_request(request_dict, operation_model)
/usr/local/lib/python3.7/site-packages/botocore/endpoint.py:198: in _send_request
    request = self.create_request(request_dict, operation_model)
/usr/local/lib/python3.7/site-packages/botocore/endpoint.py:137: in create_request
    operation_name=operation_model.name,
/usr/local/lib/python3.7/site-packages/botocore/hooks.py:412: in emit
    return self._emitter.emit(aliased_event_name, **kwargs)
/usr/local/lib/python3.7/site-packages/botocore/hooks.py:256: in emit
    return self._emit(event_name, kwargs)
/usr/local/lib/python3.7/site-packages/botocore/hooks.py:239: in _emit
    response = handler(**kwargs)
/usr/local/lib/python3.7/site-packages/botocore/signers.py:105: in handler
    return self.sign(operation_name, request)
/usr/local/lib/python3.7/site-packages/botocore/signers.py:189: in sign
    auth.add_auth(request)
/usr/local/lib/python3.7/site-packages/botocore/auth.py:418: in add_auth
    raise NoCredentialsError()
E   botocore.exceptions.NoCredentialsError: Unable to locate credentials
```

**Pull request merged**
